### PR TITLE
Handle SIGHUP and rotate log files

### PR DESCRIPTION
**Pull request by bbrks**
Updates the SIGHUP handler to rotate the new log files

### CODE REVIEW

File: base/logging.go [hunk 1]
```diff
@@ -803,6 +803,28 @@ var (
 	debugLogger, infoLogger, warnLogger, errorLogger *FileLogger
 )
 
+// RotateLogfiles rotates all active log files.
+func RotateLogfiles() error {
+	Infof(KeyAll, "Rotating log files...")
+
+	// Even if one errors, we still want to try and rotate the rest, hence the weird error handling.
+	var err error
```
Comment (tleyden) [inline base/logging.go]: Could this be a slice of `FileLoggers` that you could then iterate over the slice and call `Rotate()` on each one?  

Also, it seems odd to keep clobbering the error .. I would think if there is one error, maybe abort early

**Review by tleyden — CHANGES_REQUESTED** [review]
See inline

File: base/logging.go (hunk 1 above)
Comment (bbrks) [reply base/logging.go]: @tleyden what do you think of the new approach I've just pushed up?

File: rest/config.go [hunk 2]
```diff
@@ -908,8 +908,8 @@ func RunServer(config *ServerConfig) {
 }
 
 func HandleSighup() {
-	if err := base.RotateLogfiles(); err != nil {
-		base.Warnf(base.KeyAll, "Error rotating log files: %v", err)
+	for logger, err := range base.RotateLogfiles() {
+		base.Warnf(base.KeyAll, "Error rotating %v: %v", logger, err)
```
Comment (tleyden) [inline rest/config.go]: Doesn't this need an `if err != nil` check?

File: base/logging.go [hunk 3]
```diff
@@ -804,25 +804,21 @@ var (
...
 // RotateLogfiles rotates all active log files.
-func RotateLogfiles() error {
+func RotateLogfiles() map[*FileLogger]error {
 	Infof(KeyAll, "Rotating log files...")
 
-	// Even if one errors, we still want to try and rotate the rest, hence the weird error handling.
-	var err error
-	if e := debugLogger.Rotate(); e != nil {
-		err = e
+	loggers := map[*FileLogger]error{
```
Comment (tleyden) [inline base/logging.go]: I think this could just be initialized to an empty map

File: rest/config.go (hunk 2 above)
Comment (tleyden) [reply rest/config.go]: Will this print a friendly name for the logger?  eg, `errorLogger`?

**Review by tleyden — CHANGES_REQUESTED** [review]
see inline

File: base/logging.go (hunk 3 above)
Comment (bbrks) [reply base/logging.go]: I need something to iterate over, there's no iterable list of file loggers anywhere else unfortunately.

**Review by tleyden — APPROVED** [review]
LGTM

**Pull request merged**